pallets/runtime/common/src/bridge.rs

```diff
@@ -425,83 +426,110 @@ decl_module! {
         }
 
         /// Forces handling a transaction by bypassing the bridge limit and timelock.
-        #[weight = SimpleDispatchInfo::FixedOperational(20_000)]
+        #[weight = SimpleDispatchInfo::FixedOperational(250_000)]
         pub fn force_handle_bridge_tx(origin, bridge_tx: BridgeTx<T::AccountId, T::Balance>) -> DispatchResult {
             // NB: To avoid code duplication, this uses a hacky approach of temporarily whitelisting the did
             let sender = ensure_signed(origin)?;
-            ensure!(sender == Self::admin(), Error::<T>::BadAdmin);
-            if let Some(did) = T::CddChecker::get_key_cdd_did(&AccountKey::try_from(bridge_tx.recipient.clone().encode())?) {
-                if !Self::bridge_whitelist(did) {
-                    // Whitelist the did temporarily
-                    <BridgeLimitWhitelist>::insert(did, true);
-                    Self::handle_bridge_tx_now(bridge_tx, false)?;
-                    <BridgeLimitWhitelist>::insert(did, false);
-                } else {
-                    // Already whitelisted
-                    return Self::handle_bridge_tx_now(bridge_tx, false);
-                }
-            } else {
-                return Err(Error::<T>::NoValidCdd.into());
+            Self::force_handle_signed_bridge_tx(&sender, bridge_tx)
```

Comment:
Do we not need:
`ensure!(sender == Self::admin(), Error::<T>::BadAdmin);`

Reply:
Well spotted!

Reply:
Actually it's in `force_handle_signed_bridge_tx`.

Reply:
@maxsam4 Why is it more optimal?

Reply:
In batch functions, if the check is in the internal function, the check will be done N times. If the check is outside, it will be done just 1 time.